widget.urs

```diff
@@ -79,6 +79,22 @@ type opt_timebox
 type opt_timebox_config
 val opt_timebox : t (option time) opt_timebox opt_timebox_config
 
+type datebox
+type datebox_config
+val datebox : t time datebox datebox_config
+
+type opt_datebox
+type opt_datebox_config
+val opt_datebox : t (option time) opt_datebox opt_datebox_config
```

Comment:
Over-riding `Time` for dates is definitely going to bite us later.  We _do_ need to make a `date` newtype, but let's punt for now.